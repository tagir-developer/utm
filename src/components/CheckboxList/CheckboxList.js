import React, { Fragment } from 'react'
import Checkbox from '../Checkbox/Checkbox'

const CheckboxList = ({ checkboxes, checkedSource, changeCheckedSource }) => {

	const checkboxesList = (classes, device) => (
		checkboxes.map((checkbox, index) => (
			<div key={device + index} className={classes}>
				<Checkbox 
					checkbox={checkbox} 
					checkedSource={checkedSource} 
					index={index} 
					device={device}
					changeCheckedSource={changeCheckedSource}
					/>
			</div>
		))
	)


	return (
		<Fragment>
			<div className="d-none d-sm-block">
				<div className="row row-cols-4">
					{checkboxesList('col-3 pl-3 py-3', 'PC')}
				</div>
			</div>
			{/* For mobile devices */}
			<div className="d-block d-sm-none">
				<div className="row row-cols-2">
					{checkboxesList('col-6 pl-3 py-3', 'Mobile')}
				</div>
			</div>
		</Fragment>
	)
}



CheckboxList.defaultProps = {
	checkboxes: ['1', '2', '3'],
	checkedSource: '1',
	changeCheckedSource: () => {}
}

export default CheckboxList