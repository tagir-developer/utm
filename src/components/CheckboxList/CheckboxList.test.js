import React from 'react'
import CheckboxList from './CheckboxList'


describe('<CheckboxList />', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<CheckboxList />)
	})

	describe('Component render. Component props testing', () => {

		it('Should CheckboxList component render without props', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})


		it('Should render custom checkboxes', () => {
			wrapper.setProps({
				checkboxes: ['test-1', 'test-2', 'test-3']
			})
			expect(wrapper.find('Checkbox').at(0).props().checkbox).toBe('test-1')
			expect(wrapper.find('Checkbox').at(2).props().checkbox).toBe('test-3')
		})

		it('Should render custom checkedSource', () => {
			wrapper.setProps({
				checkedSource: 'test-1'
			})
			expect(wrapper.find('Checkbox').at(0).props().checkedSource).toBe('test-1')
			expect(wrapper.find('Checkbox').at(2).props().checkedSource).toBe('test-1')
		})

		it('Should render right amount of Checkbox component', () => {
			wrapper.setProps({
				checkboxes: ['1', '2', '3', '4']
			})
			expect(wrapper.find('Checkbox')).toHaveLength(8)
		})

	})

	describe('CheckboxList component handlers testing', () => {

		it('Should use default handler', () => {
			const result = CheckboxList.defaultProps.changeCheckedSource()
			expect(result).toBe(undefined)
		})

	})


})