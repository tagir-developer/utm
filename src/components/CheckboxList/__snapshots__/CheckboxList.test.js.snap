// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxList /> Component render. Component props testing Should CheckboxList component render without props 1`] = `
<Fragment>
  <div
    className="d-none d-sm-block"
  >
    <div
      className="row row-cols-4"
    >
      <div
        className="col-3 pl-3 py-3"
        key="PC0"
      >
        <Checkbox
          changeCheckedSource={[Function]}
          checkbox="1"
          checkedSource="1"
          device="PC"
          index={0}
        />
      </div>
      <div
        className="col-3 pl-3 py-3"
        key="PC1"
      >
        <Checkbox
          changeCheckedSource={[Function]}
          checkbox="2"
          checkedSource="1"
          device="PC"
          index={1}
        />
      </div>
      <div
        className="col-3 pl-3 py-3"
        key="PC2"
      >
        <Checkbox
          changeCheckedSource={[Function]}
          checkbox="3"
          checkedSource="1"
          device="PC"
          index={2}
        />
      </div>
    </div>
  </div>
  <div
    className="d-block d-sm-none"
  >
    <div
      className="row row-cols-2"
    >
      <div
        className="col-6 pl-3 py-3"
        key="Mobile0"
      >
        <Checkbox
          changeCheckedSource={[Function]}
          checkbox="1"
          checkedSource="1"
          device="Mobile"
          index={0}
        />
      </div>
      <div
        className="col-6 pl-3 py-3"
        key="Mobile1"
      >
        <Checkbox
          changeCheckedSource={[Function]}
          checkbox="2"
          checkedSource="1"
          device="Mobile"
          index={1}
        />
      </div>
      <div
        className="col-6 pl-3 py-3"
        key="Mobile2"
      >
        <Checkbox
          changeCheckedSource={[Function]}
          checkbox="3"
          checkedSource="1"
          device="Mobile"
          index={2}
        />
      </div>
    </div>
  </div>
</Fragment>
`;
